listeners.py

```diff
@@ -50,7 +57,48 @@ def process_result(self, obj):
                                       obj.result_checksum,
                                       obj.show_reproduce_content))
 
+    def get_failed_task_name(self, task):
+        return task[0][0] + ((':' + task[0][1]) if task[0][1] else '')
+
+    def get_long_mark(self, task):
+        return '(long)' if task in self.long_tasks else ''
+
     def print_statistics(self):
+        field_size = 60
+        top_rss = 10
+        results_sorted = []
+
+        # prepare standalone subpath '<vardir>/statistics' for statistics files
+        stats_dir = os.path.join(Options().args.vardir, 'statistics')
+        safe_makedirs(stats_dir)
+
+        # RSS
+
+        # print to stdout RSS statistics for all failed tasks
+        if self.failed_tasks:
+            color_stdout('RSS of the failed tasks:\n', schema='test_var')
+            for task in self.failed_tasks:
+                task_name = self.get_failed_task_name(task)
+                if task_name in self.rss_results:
+                    color_stdout('* %6.1f %s %s\n' % (float(self.rss_results[task_name]) / 1024,
+                                 task_name.ljust(field_size), self.get_long_mark(task_name)),
+                                 schema='test_var')
+
+        # print to stdout RSS statistics for some number of most it used tasks
+        color_stdout('Up to {} most RSS used tasks in Mb:\n'.format(top_rss), schema='test_var')
+        results_sorted = sorted(self.rss_results.items(), key=lambda x: x[1], reverse=True)
+        for task_name, rss in results_sorted[:top_rss]:
+            color_stdout('* %6.1f %s %s\n' % (float(rss) / 1024, task_name.ljust(field_size),
+                         self.get_long_mark(task_name)), schema='test_var')
+        color_stdout('-' * 41, "\n", schema='separator')
+
+        # print RSS statistics to '<vardir>/statistics/rss.log' file
```

Comment:
We should copy this directory into artifacts in `ArtifactsWatcher` I think.

Reply:
I mean the whole directory with statistics.

Reply:
Ok, sure done.